webtau-rest/src/main/java/com/twosigma/webtau/http/HttpUrl.java

```diff
@@ -42,6 +46,34 @@ public static String extractPath(String url) {
         }
     }
 
+    public static Map<String, List<String>> extractQueryParams(String url) {
+        String queryParams;
+        if (!isFull(url)) {
+            queryParams = extractQueryParamsFromRelativeUrl(url);
```

Comment:
most likely subjective, but what about declaring and assigning variable in one statement?

Reply:
You mean with a tertiary operator?

Reply:
yes. hence subjective :) no Kotlin if-else option unfortunately. 
Another option is to extract to a function and do if-else-return. 
I have fear of unassigned variable after C++.

Reply:
I do too but with a local var like here, the compiler will throw an error for unassigned var usage.